### PR TITLE
Fix handling of crashkernel parameter for Xen

### PR DESCRIPTION
**Pull request by imobach**
It fixes [bsc#962008](https://bugzilla.suse.com/show_bug.cgi?id=962008).

### CODE REVIEW

**Comment by wfeldt** [comment]
Looks much more logical than before. I think this is ok.

File: src/Kdump.ycp [hunk 1]
```diff
@@ -280,6 +280,15 @@ global void SetModified() {
...
 }
 
+/**
+ * Determine whether boot section is Xen or not
+ * It relies on section_pos
+ *
+ * @return boolean
+ */
+boolean is_xen () {
+	return (section_pos != -1);
```
Comment (lslezak) [inline src/Kdump.ycp]: NP: comparing to -1 is OK, but in such situations I rather tend to write `> 0` to handle also other (though unlikely) negative values. It's more robust and reliable...

File: src/Kdump.ycp [hunk 2]
```diff
@@ -981,15 +1015,11 @@ boolean WriteKdumpBootParameter () {
...
 	{
             result = Bootloader::setKernelParam (actual_boot_section, "crashkernel", crash_value);
-	} else {
-	    BootCommon::sections[section_pos, "xen_append"]= "crashkernel="+convertCrashkernelForXEN(crash_value);
-	    // added flag which means that section was changed bnc #432651
-	    BootCommon::sections[section_pos, "__changed"] = true;
-	    result = true;
-	    y2milestone("Write boot section to XEN boot section %1", BootCommon::sections[section_pos]:nil);
+	} else { // boot section is Xen
+		result = UpdateXenAppendParameter("crashkernel", convertCrashkernelForXEN(crash_value));
```
Comment (lslezak) [inline src/Kdump.ycp]: `UpdateXenAppendParameter()` checks for `is_xen()` internally as well, you can make the code even smaller, no need to duplicate the check. (I'd just adjust the logged message to `skipping to update Xen boot section` as for non-Xen it's not a problem.)

File: src/Kdump.ycp [hunk 3]
```diff
@@ -940,6 +950,30 @@ boolean addCrashkernelToKexec (string crashkernel_value)
...
+		y2milestone("Not Xen: refusing to update Xen boot section");
+		return false;
+	} else {
+		string xen_append = BootCommon::sections[section_pos, "xen_append"]:"";
+		BootCommon::sections[section_pos, "xen_append"] =
+			BootCommon::setKernelParamToLine(xen_append, key, value);
+		// added flag which means that section was changed bnc #432651
+		BootCommon::sections[section_pos, "__changed"] =
+			(BootCommon::sections[section_pos, "xen_append"]:"" != xen_append);
+		y2milestone("Write boot section to Xen boot section %1", BootCommon::sections[section_pos]:nil);
```
Comment (lslezak) [inline src/Kdump.ycp]: NP: The message seems a bit confusing to me, a simple `Using Xen boot section: %1` would be better IMHO.

**Comment by lslezak** [comment]
Just few minor issues, otherwise LGTM.

File: src/Kdump.ycp (hunk 2 above)
Comment (imobach) [inline src/Kdump.ycp]: Yes, you're right. I only wanted to make sure that UpdateXenAppendParameter was not called accidentally (just some kind of guard clause).

File: src/Kdump.ycp (hunk 1 above)
Comment (imobach) [inline src/Kdump.ycp]: It makes sense. I'm changing to `>= 0` in this case.

**Comment by lslezak** [comment]
The PR is open against `Code-11-SP3` but the bug report is open for SLE11-SP4, do we really fix also in SP3?

**Comment by imobach** [comment]
I've branched SLE-11-SP4 and I've created a new PR #59 against that branch.

**Pull request closed without merging**